lib/pavilion/test_config/file_format.py

```diff
@@ -19,6 +19,19 @@ class TestConfigError(ValueError):
 VAR_NAME_RE = re.compile(r'^[a-zA-Z][a-zA-Z0-9_-]*[\?\+]?$')
 
 
+class FlexibleKeyElem(yc.CategoryElem):
+    """This is for category elements that need flexible key regex."""
+
+    _NAME_RE = re.compile(r"^[\w\-. ]+$")
+
+    # TODO: Not sure what to alter here.
+    def __init__(self, name=None, **kwargs):
+        super(FlexibleKeyElem, self).__init__(name=name,
+                                              sub_elem=yc.ListElem(),
+                                              defaults=None,
+                                              **kwargs)
+
+
```

Comment:
Not sure what to do with this; specifically I'm not sure how I can override `CategoryElem` to give me a list of dictionaries where the `key` is the file name and the `value` is a list of strings (body of a file).

Any advice @Paul-Ferrell ?

Reply:
You don't have to modify __init__ at all.

I'd name it 'PathCategory', set the _NAME_RE to something that will match paths, and that's it. 
You'll also want to preserve the case of the keys. To do that, when you instantiate it, set key_case=PathCategory.KC_MIXED